docs/release/release-process.md

```diff
@@ -0,0 +1,116 @@
+# Release Management Process
+
+## Table of Contents
+
+1. [Feature Freeze & Release Branch Creation](#1-feature-freeze--release-branch-creation)
+2. [Changes to a Release Branch](#2-changes-to-a-release-branch)
+   - [Change Policy](#change-policy)
+   - [Change Process](#change-process)
+3. [Tagging a Release](#3-tagging-a-release)
+4. [Github Release](#4-github-release)
+   - [Creating the Release](#creating-the-release)
+   - [Release Notes](#release-notes)
+
+---
+
+### 1. **Feature Freeze & Release Branch Creation**
+
+Enacting a feature freeze helps to ensure that the code we publish to production environments is well tested and
+mature. The start of a feature freeze is marked by the creation of a release branch, which allows development
+against `master` to continue uninterrupted while the release is prepared.
+
+#### Plan Feature Freeze
+
+- A feature freeze may be tied either to a date scheduled in advance, or to the completion of a key feature.
```

Comment:
What do people think about releasing on a fixed cadence? Say every 2 weeks to match the code freeze of the previous release. So a new code-freeze/testing period starts as soon as a new release is made. Or maybe once per month if we find this cadence too fast?